src/win/win_ui.c

```diff
@@ -349,15 +348,22 @@ MainWindowProcedure(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
 				SendMessage(hwndSBAR, SB_GETBORDERS, 0, (LPARAM) sb_borders);
 
 				GetWindowRect(hwnd, &rect);
-				GetClientRect(hwnd, &rc);
-				non_client_width = (rect.right-rect.left) - (rc.right-rc.left);
-				non_client_height = (rect.bottom-rect.top) - (rc.bottom-rc.top);
-
 				/* Main Window. */
-				MoveWindow(hwnd, rect.left, rect.top,
-					unscaled_size_x + non_client_width,
-					unscaled_size_y + non_client_height + sbar_height,
-					TRUE);
+				if (GetSystemMetrics(SM_CXPADDEDBORDER) == 0) {
+					/* For platforms that subsystem version < 6.0 (default on mingw/msys2) */
+					/* In this case, border sizes are different between resizable and non-resizable window */
+					MoveWindow(hwnd, rect.left, rect.top,
```

Comment:
If Windows XP support is not considered, we can just force the subsystem version to 6.0 and get rid of this nightmare...

Reply:
I merged your pull request now, but yes, I do think we should force the subsystem version to 6.0, as I don't really think supporting Windows XP makes sense anymore.